### PR TITLE
chore: Add a fallback UUID for usage pulse calls

### PR DESCRIPTION
**Pull request by sbalaji1192**
## Description
Added a fallback value (crypto.randomUUID) when there is an issue with generating the key from segment

Fixes #https://github.com/appsmithorg/cloud-services/issues/582

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update

- Manual
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by berzerkeer — APPROVED** [review]
nit: any reason to add a polyfill and use crypto to generate random uuid when we have `nanoId` package ?

**Comment by sbalaji1192** [comment]
@berzerkeer Crypto randomUUID should suffice for all modern browsers. I was adding the polyfill as a fail-safe in older browsers. I didn't look into nano UUID, so as not to bloat the bundle.

**Comment by berzerkeer** [comment]
> @berzerkeer Crypto randomUUID should suffice for all modern browsers. I was adding the polyfill as a fail-safe in older browsers. I didn't look into nano UUID, so as not to bloat the bundle.

Okay, asked because we already have nanoId as a dependency in our repo.

**Comment by sbalaji1192** [comment]
Oh. then in that case lets use nanoID. didn't know that.

**Review by berzerkeer — DISMISSED** [review]

File: app/client/src/ce/sagas/userSagas.tsx [hunk 1]
```diff
@@ -147,6 +148,18 @@ export function* initiateUsageTracking(payload: {
     } else {
       yield initializeSegmentWithoutTracking();
```
Comment (sbalaji1192) [inline app/client/src/ce/sagas/userSagas.tsx:149]: 1. remove this line and corresponding functions.
2. when telemetry is enabled, we need to fetch the anonymous in every sendPulse of usage pulse file.

**Pull request closed without merging**